### ISSUE TITLE
Supplying region should be optional

### ISSUE THREAD

**Issue by andrewlytle** [aws-actions/configure-aws-credentials#305]
I'm using this action to assume a role for CFN deployments. In the case where I might want to use the tool to deploy to multiple regions, I would need to update my action for each region I wish to deploy into. I'd rather embed this in my deployment tool config than the github action.

An explicit Region is not required when assuming a role, correct? Would it be reasonable to request making this optional? It seems like it is mostly used to get a Regional STS endpoint?

**Comment by ultimaweapon** [comment]
We plan to deploy our application on multiple regions too. One of the obstacle we have right now is this issue.

**Comment by niceFont** [comment]
I would like to work on this. Would it be possible to assign this to me?

**Comment by ohcnkk** [comment]
Please also enable to choose global endpoint.
I am facing an issue where the actions needs to authenticate with global endpoint instead of regional endpoint.

**Comment by peterwoodworth** [comment]
I don't think this is supported in js v3 yet, which will prevent our adoption of this https://github.com/aws/aws-sdk-js-v3/issues/4568

**Comment by onkar-indellient** [comment]
are there any updates on this?

**Comment by kellertk** [comment]
AWS is moving to make regional STS endpoints required in the future, so therefore they'll be required here as well. This aligns with AWS best practice of not relying on global endpoints if possible because they are not as resilient to failure as the regional endpoints. 

We use `core.getInput` [from the Actions toolkit](https://github.com/actions/toolkit/blob/1b1e81526b802d1d641911393281c2fb45ed5f11/packages/core/src/core.ts#L151) to get the region. In the runner environment, inputs are constructed from environment variables, in this case `INPUT_AWS-REGION`. If you wish to run this Action multiple times with different regions, you might be able to set this environment variable yourself. Alternatively, you could run the Action multiple times in different workflows: one for each region.

If there's significant input from the community, we can reconsider this, but for now we don't plan to change the required-ness of the AWS region.
